### DISCUSSION TITLE
How to write complex enums?

### DISCUSSION THREAD

**Discussion by demfabris** [tokio-rs/prost#596] [Q&A]
How to write this rust enum in protobuf?

```rust
pub enum Instruction {
    Provide { hash: String, data: Gistit },
    Fetch { hash: String },
    Status,
    Shutdown,
    Response(ServerResponse),
}
```

**Comment by LucioFranco** [comment]
You will want to combine `oneof` and messages to achieve the rich enum variants. Othrwise you can just follow the regular protobuf enum support.
